lib/swaggerValidator/semanticValidator.ts

```diff
@@ -346,12 +347,49 @@ export class SwaggerSemanticValidator {
           property: d,
         });
         this.addErrorsFromErrorCode(errors, rootInfo.url, meta, sch);
-      } else if (sch.properties[d].type !== "string") {
-        const meta = getOavErrorMeta("DISCRIMINATOR_PROPERTY_TYPE_NOT_STRING", {
-          property: d,
-        });
-        this.addErrorsFromErrorCode(errors, rootInfo.url, meta, sch);
+      } else {
+        const discriminatorProp = jsonLoader.resolveRefObj(sch.properties[d]);
+        if (discriminatorProp.type !== "string") {
+          const meta = getOavErrorMeta("INVALID_DISCRIMINATOR_TYPE", {
+            property: d,
+          });
+          this.addErrorsFromErrorCode(errors, rootInfo.url, meta, sch);
```

Comment:
Do we still need validate further, if already find INVALID_DISCRIMINATOR_TYPE error.